adapters/repos/db/index_sharding_backward_compatibility.go

```diff
@@ -33,7 +32,7 @@ func (i *Index) checkSingleShardMigration(shardState *sharding.State) error {
 		}
 
 		// whatever is left now, needs to be migrated
-		shards := shardState.AllPhysicalShards()
+		shards := i.shardState.AllPhysicalShards()
```

Comment:
I'm not 100% sure about moving the sharding state into the Index struct itself. Is Index's sharding state updated no somewhere if it changes?

Reply:
the shardState was already used by the index i.e. initialize the shards, knowing if partitioning was enabled. The reason I keep a reference in the struct is because when async-replication is restarted we need to know the physical shards.
the shardState is immutable, a copy is used when the index is constructed and shards are loaded based on its information.